core/tracker/inclusion.go

```diff
@@ -0,0 +1,325 @@
+		i.missedFunc(ctx, sub)
+	}
+	i.submissions = remaining
+}
+
+// CheckBlock checks whether the block includes any of the submitted duties.
+func (i *inclusionCore) CheckBlock(ctx context.Context, block block) {
+	i.mu.Lock()
+	defer i.mu.Unlock()
+
+	var remaining []submission
+	for _, sub := range i.submissions {
+		switch sub.Duty.Type {
+		case core.DutyAttester, core.DutyAggregator:
+			_, ok := block.Attestations[sub.AttRoot]
+			if !ok {
+				remaining = append(remaining, sub)
+				continue
+			}
+
+			// TODO(corver): We should probably check if validator included in AggregationBits
+			//  for attester duty by caching scheduler duty data to get validator committee index.
+
+			// Report included and trim
+			i.attIncludedFunc(ctx, sub, block)
+		case core.DutyProposer, core.DutyBuilderProposer:
+			if sub.Duty.Slot != block.Slot {
+				remaining = append(remaining, sub)
+				continue
+			}
+
+			// Nothing to report for block inclusions, just trim
+		default:
+			panic("bug: unexpected type") // Sanity check, this should never happen
+		}
+	}
+
+	i.submissions = remaining
+}
+
+// reportMissed reports duties that were broadcast but never included on chain.
+func reportMissed(ctx context.Context, sub submission) {
+	inclusionMisses.WithLabelValues(sub.Duty.Type.String()).Inc()
+
+	switch sub.Duty.Type {
+	case core.DutyAttester, core.DutyAggregator:
+		msg := "Broadcasted attestation never included on-chain"
+		if sub.Duty.Type == core.DutyAggregator {
+			msg = "Broadcasted attestation aggregate never included on-chain"
+		}
+
+		log.Warn(ctx, msg, nil,
+			z.Any("pubkey", sub.Pubkey),
+			z.I64("attestation_slot", sub.Duty.Slot),
+			z.Any("broadcast_delay", sub.Delay),
+		)
+	case core.DutyProposer, core.DutyBuilderProposer:
+		msg := "Broadcasted block never included on-chain"
+		if sub.Duty.Type == core.DutyBuilderProposer {
+			msg = "Broadcasted blinded block never included on-chain"
+		}
+
+		log.Warn(ctx, msg, nil,
+			z.Any("pubkey", sub.Pubkey),
+			z.I64("block_slot", sub.Duty.Slot),
+			z.Any("broadcast_delay", sub.Delay),
+		)
+	default:
+		panic("bug: unexpected type") // Sanity check, this should never happen
+	}
+}
+
+// reportAttInclusion reports attestations that were included in a block.
+func reportAttInclusion(ctx context.Context, sub submission, block block) {
+	att := block.Attestations[sub.AttRoot]
+	aggIndices := att.AggregationBits.BitIndices()
+	attSlot := int64(att.Data.Slot)
+	blockSlot := block.Slot
+	inclDelay := block.Slot - attSlot
+
+	msg := "Broadcasted attestation included on-chain"
+	if sub.Duty.Type == core.DutyAggregator {
+		msg = "Broadcasted attestation aggregate included on-chain"
+	}
+
+	log.Info(ctx, msg,
+		z.I64("block_slot", blockSlot),
+		z.I64("attestation_slot", attSlot),
+		z.Any("pubkey", sub.Pubkey),
+		z.I64("inclusion_delay", inclDelay),
+		z.Any("broadcast_delay", sub.Delay),
+		z.Int("aggregate_len", len(aggIndices)),
+		z.Bool("aggregated", len(aggIndices) > 1),
```

Comment:
@dB2510 the aggregated logging field is added here

Reply:
Just to clarify, this aggregated logging will tell us that attestation is an aggregate attestation and NOT an unaggregated one. Well here `aggregated: true` only when we are aggregator of that slot AND it is our aggregate attestation.